### ISSUE TITLE
No Unbox instances for newtypes in Data.Monoid

### ISSUE THREAD

**Issue by Boarders** [haskell/vector#264]
Currently you cannot use an Unboxed vector with Any, All, Sum, Product. I think there should be instances of the form:

```haskell
instance (Unbox a) => instance (Unbox (Sum a))
instance (Unbox a) => instance (Unbox (Product a))
-- etc
```
These should be defined in `vector` and not in `Data.Monoid` as this typeclass is a particular quirk of how the vector library works.

**Comment by chessai** [comment]
I agree that these instances should exist.

Do note that they couldn't exist in Data.Monoid because base is a dependency of vector.

Thanks

**Comment by fumieval** [comment]
They've been added to master a while ago (https://github.com/haskell/vector/commit/bf3b7cd544c8aad7ec0c88ae3df822681384a6b2) , just not available on Hackage yet (#265).

**Comment by lehins** [comment]
This is a common pattern. I've raised this issue in #258 already and still hoping for the release and development process of `vector` to get fixed. All of these tickets exist only because fixes get cherry picked from master prior to a release:
* #255
* #256
* #258
* #265
* Now this one. And who know how many more fixes we are missing in the released versions that we would expect those issues to be fixed.

**Comment by cartazio** [comment]
@lehins  could you list the tickets on the release tick i kicked off so i can keep track of what i will include or not in this coming release (had a fever this past week etc  etc, so catching up on this backlog)

**Comment by Shimuuar** [comment]
Fixed since 0.12.1 release
